### ISSUE TITLE
Rhythm Instructional Aids

### ISSUE THREAD

**Issue by rpspringuel** [gregorio-project/gregorio#1173]
One of the things I was exposed to in my class on chant the week before last were some instructional diagrams designed to help people learn the rhythm of Gregorian chant.  These diagrams took the form of a 1-line staff (some times with the staff line being invisible) and never used clefs or ledger lines.  Indeed, most of the time only a single note position was needed (the one on the line), but I saw at least one diagram with a podatus and another with a clivis, so clearly at least two (probably three) note positions would need to be supported.

Along similar lines, when doing a rhythmic analysis of a chant piece, we would insert 1/8th and 1/4th rests into the staffs (generally after bar-lines, but also at the beginning of the score) and an instructor indicated to me that it would be useful if these could be added into the score directly rather than drawn in by hand later (with an TeX option to shut them off).

Finally, as part of a rhythmic analysis of a score, the instructors add vertical episemata (did I get that plural right this time?) to show where all the icti are.  It would thus be useful if we could have an optional (instructional) vertical episemus (or perhaps more than one) which could be turned on and off in the TeX file (enabling both a finished score and a notated one to be produced from the same gabc file).  I won't go so far as to suggest that we should be able to automatically determine the location of all icti, as there are some occasions which are ambiguous (and thus left to the discretion of the choir master), but there are some rules (which I don't have in front of me but can post later) which indicate certain places as being required to have an ictus.  Thus a three stage system (only explicit {currently `'` in gabc}, explicit+automatic, explicit+automatic+instructional) would be useful.

**Comment by rpspringuel** [comment]
An example of the 1-line staff:

![temp](https://cloud.githubusercontent.com/assets/7671682/16565870/4f46478e-41de-11e6-8d90-8b19f2d53b5b.png)

The curved lines above are chironomy markings and can be ignored so far as this feature request is concerned.

**Comment by rpspringuel** [comment]
Another example (this time with the staff line invisible):
![scan](https://cloud.githubusercontent.com/assets/7671682/16566852/49774d1e-41e6-11e6-8a7f-5f922337b460.jpeg)

Again, the curved lines are chironomy markings and can be ignored so far as this feature request is concerned.

**Comment by rpspringuel** [comment]
The rules for determining the placement of an ictus:

1. on all notes given in the text with the printed vertical episema (already available)
2. on all long notes (i.e. dotted notes, the first note of a distropha, tristropha, pressus, oriscus at the unison, and first of the notes of "fusion")
3. on the note that precedes the quilisma
4. on the vigra at the top of a note-group, unless otherwise indicated
5. on the first note of a neum
6. if a melody begins on an up-pulse, a single-pulse rest is added at the start to complete the binary rhythm.  This added rest is ictic.
7. In syllabic chant, spondees and dactyls are given their natural rhythm, otherwise count by two's in retrograde motion from the first known ictus.

I think rules 1-5 are candidates for some level of automation.  An unmentioned rule here is that all rhythms in Gregorian chant are either binary or ternary so icti cannot appear on adjacent notes unless the first note is a lengthened one.  This is why rule 4 says "unless otherwise indicated" (and really rules 3 and 5 should say the same).

**Comment by henryso** [comment]
Well, those rules really only apply if you're using the Ward method of counting twos and threes rather than more evolved methods brought on by the semiology research spearheaded by Solesmes's Dom Cardine, so I wouldn't make automate any rules in Gregorio proper; perhaps in an external tool.

**Comment by eroux** [comment]
I agree with @henryso on this: Gregorio is mainly about typesetting things, with some options. For complex constructions not directly linked with typesetting (icti automation mainly), I don't think it's Gregorio's role... You can start a new repository in the organizaion though, no problem

**Comment by rpspringuel** [comment]
Ah, okay.  I didn't realize there were other methods for this.  Seems my lack of knowledge is showing through.

So forget the automatic ictus placement stuff.  The other diagramatic stuff, however, should still be useful (and relatively independent of method).

**Comment by eroux** [comment]
I'm not entirely sure I understand the state of the request... @rpspringuel maybe you could rewrite what you need more explicitely? Currently it's possible to switch the episemata on and off, what else do you need? Do you need some way to indicate optional episemata in gabc, that would be triggered by a switch in TeX?

**Comment by rpspringuel** [comment]
Okay, to summarize what this request is all about:

1) 1-line staff (we currently cover 2 to 5 lines)
2) instructional 1/8 and 1/4 note rests
3) instructional episemata

Instructional items are entered in the gabc but not shown by default.  An option makes them visible (and in the case of the rests, adjusts spacing to account for them).
